src/data/AutoScroll.js

```diff
@@ -300,8 +300,13 @@ chrome.storage.local.get(defaults, function (options) {
   function isInvalid(elem) {
     return elem.isContentEditable ||
            (elem.localName === "a" && elem.href) ||
-           (elem.localName === "textarea") ||
-           (elem.localName === "input");
+           (elem.localName === "area" && elem.href) ||
+           (elem.localName === "textarea") && isEditableText(elem) ||
+           (elem.localName === "input") && isEditableText(elem);
```

Comment:
Please add parentheses around the `textarea` and `input` lines, so that it's clear what the precedence is.